File: mpp/lab7/client/src/main/java/domain/Participant.java
package domain;

import grpc.triathlon.ParticipantProto;

import java.io.Serializable;

public class Participant implements Serializable {
    private int id = 0;
    private final String name;

    public Participant(String name) {
        this.name = name;
    }

    public Participant(int id, String name) {
        this(name);

        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public static Participant fromProto(ParticipantProto proto) {
        return new Participant(proto.getId(), proto.getName());
    }

    @Override
    public String toString() {
        return "Participant{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}

File: mpp/lab7/server/Domain/Participant.cs
using Triathlon;

namespace server.Domain
{
    public class Participant
    {
        public int Id { get; set; }
        public string Name { get; }

        public Participant(int id, string name)
        {
            Id = id;
            Name = name;
        }

        public Participant(string name)
                : this(0, name)
        { }

        public override string ToString()
        {
            return $"Participant{{Id={Id}, Name={Name}}}";
        }

        public ParticipantProto ToProto()
        {
            return new ParticipantProto { Id = Id, Name = Name};
        }
    }
}
